src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -371,7 +371,7 @@ public virtual VerifyResult VerifyStateDependent(ProtocolSettings settings, Data
                 else
                 {
                     if (!this.VerifyWitness(settings, snapshot, hashes[i], witnesses[i], net_fee, out long fee))
-                        return VerifyResult.Invalid;
+                        return VerifyResult.InvalidSignature;
```

Comment:
This is not necessarily an invalid signature, right?

Reply:
We can use `Invalid` if you prefer

Reply:
Fixed